function/make.go

```diff
@@ -215,6 +223,49 @@ func MakeFunction(name string, function interface{}) MetricFunction {
 			}
 		},
 	}
+
+	for _, option := range options {
+		// TODO: check there aren't multiple conflicting values
+		if option.Name == "WidenBy" || option.Name == "ShiftBy" {
+			sign := -1
+			if option.Name == "ShiftBy" {
+				sign = 1
+			}
+			switch value := option.Value.(type) {
```

Comment:
Let's and an `else` clause here to fail if it doesn't know how to handle `option.Value.(type)`